### PR TITLE
feat: support EIP-1559 in send modal

### LINKED ISSUE #1181: Send modal doesn't support EIP-1559
The send modal doesn't plumb through the `maxFeePerGas`/`maxPriorityFeePerGas` to enable EIP-1559 TXes on wallets that support it.

### PR DESCRIPTION
**Pull request by mrnerdhair**
## Description

Adds EIP-1559 fee support to `useFormSend()`. 1559-style fees will be used to `buildSendTransaction()` if they are available and the wallet supports them; otherwise legacy `gasPrice`-style fees will be used.

(Requires shapeshift/lib#440 and an associated version bump.)

## Notice

Before submitting a pull request, please make sure you have answered the following:

- [x] Have you followed the guidelines in our [Contributing]('https://github.com/shapeshift/web/CONTRIBUTING.md) guide?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/shapeshift/web/pulls) for the same update/change?

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

Closes #1181. Possibly relevant to #1025.

## Risk

AFAIK, while 1559 fee estimates are provided by `chain-adapters`, we haven't been using them in prod yet. Incorrect fee estimations could cause problems by slowing or failing user TXes, or by wasting too much money on gas.

## Testing

For both an ETH send and an ERC-20 send:
- Test that a native wallet send TX uses EIP-1559
- Test that a KK send TX uses EIP-1559
- Test that a MM send TX uses the same value for "Max Fee" and "Max Priority Fee" (which is because EIP-1559 isn't supported by `hdwallet-keepkey` yet)
image

### CODE REVIEW

**Comment by 0xdef1cafe** [comment]
@mrnerdhair after 3 weeks can you please close this or move this out of draft for review?

**Comment by mrnerdhair** [comment]
Can't move forward until I get an owner stamp on shapeshift/lib#440

**Comment by 0xdef1cafe** [comment]
> Can't move forward until I get an owner stamp on [shapeshift/lib#440](https://github.com/shapeshift/lib/pull/440)

this is now merged, can you please bump chain adapters in this PR and get it ready for review?

**Comment by mrnerdhair** [comment]
Should be good to go now; also re-enabled a bunch of accidentally-disabled tests in `useFormSend.test.tsx` by getting rid of an `it.only`!

File: src/components/Modals/Send/hooks/useFormSend/useFormSend.tsx [hunk 1]
```diff
@@ -36,15 +37,28 @@ export const useFormSend = () => {
...
         if (adapterType === ChainTypes.Ethereum) {
+          if (!supportsETH(wallet)) throw new Error(`useFormSend: wallet does not support ethereum`)
           const fees = estimatedFees[feeType] as chainAdapters.FeeData<ChainTypes.Ethereum>
-          const gasPrice = fees.chainSpecific.gasPrice
-          const gasLimit = fees.chainSpecific.gasLimit
+          const {
+            chainSpecific: { gasPrice, gasLimit, maxFeePerGas, maxPriorityFeePerGas }
+          } = fees
           const address = isEthAddress(to) ? to : ((await ensLookup(to)).address as string)
+          const useEIP1559Fees =
```
Comment (gomesalexandre) [inline src/components/Modals/Send/hooks/useFormSend/useFormSend.tsx:46]: Nitpick: `useXYZ` is usually used for naming hooks/hook-ish variables in React. Would be happier with something like `withBaseFee` or `withEIP1559Fees`

**Review by gomesalexandre — APPROVED** [review]
Tested against develop and it does indeed bring EIP-1559 fees support in Native and KeepKey wallets, which is not present in develop.

### Native wallet

- [`support-eip-1559-in-sendmodal` Tx](https://etherscan.io/tx/0x68f03dab273db4817ed050bc6933ca9f9922f4e6917a4dac157246cb6b3c4ea6)
image


- [`develop` Tx](https://etherscan.io/tx/0xe3773dd64c167af0c400924d48b06507eaf4af4e15ea490d91e060e430a362e3)
 image

KeepKey

- [`support-eip-1559-in-sendmodal` Tx](https://etherscan.io/tx/0x156137ef827a29f78d48981434caa406488c27ca769c5789bc1ed60b37901eaf)
image

- [`develop` Tx](https://etherscan.io/tx/0xf5886099859567a8ac83c9e07d5891ef77288e7e5d3d489ed96d81eda2e655e8) 
image

### MetaMask
-  [`support-eip-1559-in-sendmodal` Tx](https://etherscan.io/tx/0x18c85dad1ade66b07b3434475fce19565ce369d3334a861a93b17f25873a72a6)
image

- [`develop` Tx](https://etherscan.io/tx/0x4459bd39b1733f0721ba94e909209d882d10d7f61a5b7ac6e155282272da2ab0)
image

@mrnerdhair Despite `hdwallet-metamask` not supporting EIP-1559 (confirmed by checking the resolution of `wallet.ethSupportsEIP1559` promise in your branch), MM still broadcasts Txs using the EIP-1559 standard for gas fees - **although not sending a max fee / max priority fee** - in both develop and on this branch.
Can you update the testing steps to reflect this caveat, to make it clear for operations what is the expected behavior?

**Comment by gomesalexandre** [comment]
@mrnerdhair CI failing on lint, will need to fix that to make it pass

File: src/components/Modals/Send/hooks/useFormSend/useFormSend.tsx (hunk 1 above)
Comment (mrnerdhair) [reply src/components/Modals/Send/hooks/useFormSend/useFormSend.tsx]: good point; changed to `shouldUseEIP1559Fees`.

**Comment by mrnerdhair** [comment]
@gomesalexandre ty, feedback addressed!

**Review by gomesalexandre — APPROVED** [review]
Testing steps looking good 🍬

**Comment by 0xdef1cafe** [comment]
tested by ops https://discord.com/channels/554694662431178782/876953436908822568/963131170088943626

**Review by 0xdef1cafe — APPROVED** [review]

**Pull request merged**